src/extensions.ts

```diff
@@ -29,7 +29,7 @@ export class InitializeTextDocumentFeature implements StaticFeature {
 
     const languageClient = this.languageClient
     async function saveFile (textDocument: vscode.TextDocument) {
-      if (documentSelector != null && vscode.languages.match(documentSelector, textDocument) > 0) {
+      if (documentSelector != null && vscode.languages.match(documentSelector, textDocument) > 0 && textDocument.uri.scheme === 'file') {
```

Comment:
What if it's not a file? Should we do nothing, silently?

Reply:
That's it. This code is here to update the file on the LSP filesystem when the file is open in monaco.

If it's not a file... let's not try to update it on the filesystem since... it's not a file